src/olympia/legacy_api/views.py

```diff
@@ -314,8 +311,9 @@ def guid_search_cache_key(guid):
                     })
                 addons_xml[key] = addon_xml
 
-    cache.set_many(dict((k, v) for k, v in addons_xml.iteritems()
-                        if k in dirty_keys))
+    if dirty_keys:
+        cache.set_many(dict((k, v) for k, v in addons_xml.iteritems()
+                            if k in dirty_keys))
```

Comment:
Very small optimization to not set anything in case there aren't any dirty keys that need updates.